### ISSUE TITLE
DI Integration Testing

### ISSUE THREAD

**Issue by gdalle** [compintell/Mooncake.jl#286]
In DI, how will we handle the change from `Array` to other tangent types in 1.11?

**Comment by willtebbutt** [comment]
I think I should probably just add some conversion functionality to ensure that DI doesn't ever have to see weird tangent types for `Array`.

**Comment by willtebbutt** [comment]
I've realised that this was a false alarm -- apologies! The reason is that I need to keep gradients of `Array`s as `Array`s in order to retain compatibility with LTS. While I've not made a decision about whether the type of the gradient w.r.t. an object is part of the public API, and therefore must not change between Julia versions, my gut is that it would be kind of made to get `Array`s on 1.10, and `MutableTangent`s on 1.11.

I'm going to close this, but re-open if it becomes a problem again.

**Comment by gdalle** [comment]
Perhaps as a smoke test you could add DI testing to your test suite?

**Comment by willtebbutt** [comment]
Good shout. I'll make a point of doing this once the 1.11 transition PR is merged.

**Comment by gdalle** [comment]
The test file should be as simple as this:

```julia
using DifferentiationInterface, DifferentiationInterfaceTest
import Mooncake

backends = [AutoMooncake(; config=nothing), AutoMooncake(; config=Mooncake.Config())]

test_differentiation(
    backends;
    excluded=SECOND_ORDER,
    logging=true,
);
```

However if you release a breaking version on Mooncake, this will fail to resolve the environment until DI updates its compatibility bounds. That's the downside of handling backend compatibility in DI itself.

**Comment by willtebbutt** [comment]
Resolves by #302
